### ISSUE TITLE
AbstractTypeMaterializer attempts to materialize Iterable

### ISSUE THREAD

**Issue by jmax01** [FasterXML/jackson-modules-base#17]
AbstractTypeMaterializer should skip Iterable.class.

Something like this perhaps?

``` java
        //current
        if (type.isContainerType() || type.isPrimitive() || type.isEnumType() || type.isThrowable() ||
                type.getRawClass() == Number.class)
        {
            return null;
        }
       Class<?> cls = type.getRawClass();

        //fix
        Class<?> cls = type.getRawClass();
        if (type.isContainerType() || type.isPrimitive() || type.isEnumType() || type.isThrowable() ||
                cls  == Number.class || cls == Iterable.class)
        {
            return null;
        }       
```

**Comment by cowtowncoder** [comment]
Plain `Iterable` definitely should not be materialized, so I can start with that check.
More challenging would be the question of subtypes, but I guess we can cross that bridge when we get there (that is, someone showing usage where it matters).

**Comment by cowtowncoder** [comment]
@jmax01 Heh. Grepping through tests, I found that #12 may have fixed this; would be in 2.7.5, and is fixed in 2.8.0 I think (and will be for sure in 2.8.1, to be released today or tomorrow).
I'll double-check it, but test looks solid.

**Comment by jmax01** [comment]
We were trying to move to 2.8.x but had to roll back 2.6.7 due to databind #1302 and TypeModifer changes broke our jdk8 Stream module.

Looking at #12, my first thought when I discovered this issue was that AbstractTypeResolvers were being called too early.

**Comment by cowtowncoder** [comment]
@jmax01 I just released 2.8.1, for what that's worth.

**Comment by cowtowncoder** [comment]
I assume this has been resolved by 2.8.1.

**Comment by jmax01** [comment]
Yes works.
